### PR TITLE
Header files

### PR DESCRIPTION
**Pull request by oOoOoOo0o**
Self-explanatory

 - done with Euan

### CODE REVIEW

**Comment by Bourkupine** [comment]
For anyone wondering the line:
`vector<void*> completedModuleCodes;` in `Student.h`
We are using a void pointer in the place of a Module pointer due to circular dependencies. As Module depends on Student, it wouldnt make sense for Student to also depend on Module. However Student must store an array of their completed modules, so as a fix we can use a void pointer, and when it comes to adding modules into their completed array we can just call the function to add completed modules with the Module pointer and essentially cast the void pointer to be a Module one.

tl;dr: Due to circular dependencies, we store completed modules as void pointers and cast them down to Module pointers when we use them.

**Review by ShaveYourEyebrows — APPROVED** [review]
Looks good, make sure to get the implementations in this branch

**Review by MartyD1 — APPROVED** [review]

**Review by carthy27 — APPROVED** [review]
get get get get get

**Comment by carthy27** [comment]
smart thanks for tldr i wouldnt have known. mwah
